src/BrainPass/BrainPass.sol

```diff
@@ -0,0 +1,343 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
+import {Counters} from "openzeppelin-contracts/contracts/utils/Counters.sol";
+import {Owned} from "solmate/auth/Owned.sol";
+import {SafeMath} from "openzeppelin-contracts/contracts/utils/math/SafeMath.sol";
+
+interface IERC20 {
+    function transferFrom(
+        address from,
+        address to,
+        uint256 amount
+    ) external returns (bool);
+}
+
+/// @title BRAIN Pass NFT
+/// @author Oleanji
+/// @notice A pass for IQ Wiki Editors
+
+contract BrainPassCollectibles is ERC721, Owned {
+    /// -----------------------------------------------------------------------
+    /// Errors
+    /// -----------------------------------------------------------------------
+
+    error MintingPaymentFailed();
+    error IncreseTimePaymentFailed();
+    error AlreadyMintedThisPass();
+    error NotTheOwnerOfThisNft();
+    error InvalidMaxTokensForAPass();
+    error PassTypeNotFound();
+    error PassMaxSupplyReached();
+    error NoEtherLeftToWithdraw();
+    error TransferFailed();
+    error DurationNotInTimeFrame();
+
+    /// -----------------------------------------------------------------------
+    ///  Inheritances
+    /// -----------------------------------------------------------------------
+
+    using SafeMath for uint256;
```

Comment:
is this needed?

Reply:
thats what handle the arthimetic operations in case of overflow/underflow sub, mul, div functions on the uint256

Reply:
there arent arthimetic overflows in latest versions of solidity